app/controllers/admin/insights_controller.rb

```diff
@@ -0,0 +1,17 @@
+module Admin
+  class InsightsController < Admin::ApplicationController
+    def ugly
+      @start_date = params[:start] || 1.week.ago
+      @end_date = params[:end] || Time.current
+      @insights = [
+        ["Posts published", Article.where("published_at > ? AND published_at < ?", @start_date, @end_date).size],
```

Comment:
@citizen428 Good callout. This also brings up something else we may want to consider: `published_at` is a `timestamp` column but we seem to be treating it as a `date`. Postgres can handle that, but it may not be what we expected. It will treat both `@start_date` and `@end_date` as `YYYY-MM-DDT00:00:00.000000Z` — the precision is 1µs rather than the full day. This is likely what we want for the start of the range (because it will give us the full day), but at the end of the range it will exclude all but the first microsecond of the day even if we use the inclusive end.

For example:

```ruby
irb(main):001:0> Article.where(published_at: 1.day.ago.to_date..Date.today).order(:created_at).
last.created_at
=> Wed, 10 Nov 2021 23:58:21.076355000 UTC +00:00
```

Treating a timestamp column as if it were equivalent to a date shows the last article published between yesterday and today was 14 hours ago.

```ruby
irb(main):002:0> Article.where("published_at::date BETWEEN ? AND ?", 1.day.ago.to_date, Date.to
day).order(:created_at).last.created_at
=> Thu, 11 Nov 2021 14:20:12.110201000 UTC +00:00
```

But casting `published_at` to a `date` column shows that it was just a few minutes ago. This seems to be closer to what we want.

The downside is the performance. We have an index on `published_at` (the `timestamp` column), but we don't have one for `published_at::date`. The only other alternative I can think of (disclosure: there may be others I haven't thought of) that _will_ use the index is to use `Date#end_of_day` provided by ActiveSupport.